identity/src/main/java/com/android/identity/GattServer.java

```diff
@@ -339,6 +342,18 @@ public void onNotificationSent(BluetoothDevice device, int status) {
             reportError(new Error("Error in onNotificationSent status=" + status));
             return;
         }
+
+        if (writingQueueTotalChunks > 0) {
```

Comment:
The conditions here don't seem necessary.

If `mWritingQueue.size()` is zero, then the `percentComplete` calculation will be `1`, right? This feels suspiciously like an overzealous optimization.

Reply:
The `writingQueueTotalChunks > 0` check here is because there is no way to guarantee that `onNotificationSent` is being called in response to a message transmission or some other sort of notification. I think probably that the only kind of notifications we are sending is message data, but it feels like an assumption we don't want to make per say.  Note that we could send this progress update right after sending, but my experience of BT on Android is that you should always wait for callbacks because under the hood there is no queuing and it can result in race conditions. So if for example, you wanted to send another message after some condition like `progress == max` something unexpected might happen like a lost frame. I'm not sure I'm super duper happy with this solution either BTW, because its also making an assumption that `writingQueueTotalChunks` always been set to zero after transmission but short of writing a whole complicated generic BT queuing mechanism this seems like a reasonable approach. Would like to hear your thoughts though...

Reply:
Ah, thanks for the explanation.

I don't see a better solution, really. At least, nothing that isn't unnecessarily complex. So this LGTM.

Please add a small comment indicating why the check is there, since I think it's not totally obvious (at least to me). `// there is no guarantee that this method is called in response to a message transmission` is probably sufficient.

Reply:
Done